fdbclient/NativeAPI.actor.cpp

```diff
@@ -3273,8 +3304,14 @@ ACTOR static Future<Void> tryCommit( Database cx, Reference<TransactionLogInfo>
 					commit_unknown_result()});
 		}
 
-		Version v = wait( readVersion );
-		req.transaction.read_snapshot = v;
+		if(!req.tagSet.present()) {
+			Version v = wait(readVersion);
```

Comment:
maybe `store(req.transaction.read_snapshot, readVersion)` as you did in the else {}?